hawkular-inventory-api/src/main/java/org/hawkular/inventory/api/model/InventoryStructure.java

```diff
@@ -499,6 +504,11 @@ void impl(Class<E> childType, RelativePath.Extender parent) {
                 blueprints.put(entityPath, entity.entity);
 
                 RelativePath parent = entityPath.up();
+                if (parent.equals(entityPath)) {
+                    //if we can no longer go up, don't add this entity as a child of its parent. It would add itself
+                    //as its own child...
```

Comment:
The inventory structure object (just that, not the actual DB) would contain such loop, where it would suggest to the callers that the entity indeed contains itself.